### ISSUE TITLE
The difference between pre_training and the discory?

### ISSUE THREAD

**Issue by Amazingren** [LuigiRiz/NOPS#3]
Hi Luigi,

Thanks for the impressive work on applying NCD to the 3D PCL segmentation.

I am curious that the difference between pre_training and the discovery in your code base. 

Is it true that the discovery.py is exactly the NOPS proposed in this paper and the pretraining is some kind of the EUMS^{*}

I am looking forward to your reply.

Bests,

**Comment by LuigiRiz** [comment]
Hi, 
main_pretrain.py performs some pre-training epochs on the architecture of NOPS considering only the heads related to base classes using ground-truth labels. Oppositely, main_discover.py jointly learns base and novel classes by merging ground-truth labels and predicted pseudo-labels. So, one can start main_discover.py using a checkpoint from main_pretrain.py or just start main_discover.py from scratch. Note that, as stated in the paper, the second option (no pretraining) leads to much better results. 

Hope I clarified your doubts

Best,

Luigi

**Comment by Amazingren** [comment]
> Hi, main_pretrain.py performs some pre-training epochs on the architecture of NOPS considering only the heads related to base classes using ground-truth labels. Oppositely, main_discover.py jointly learns base and novel classes by merging ground-truth labels and predicted pseudo-labels. So, one can start main_discover.py using a checkpoint from main_pretrain.py or just start main_discover.py from scratch. Note that, as stated in the paper, the second option (no pretraining) leads to much better results.
> 
> Hope I clarified your doubts
> 
> Best,
> 
> Luigi


Thanks a lot for the detailed clarification. Now it is much clear to me.

Bests and have a nice day!
